pkg/rbacassessment/builder.go

```diff
@@ -0,0 +1,150 @@
+package rbacassessment
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/trivy-operator/pkg/kube"
+	"github.com/aquasecurity/trivy-operator/pkg/trivyoperator"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+type ReportBuilder struct {
+	scheme           *runtime.Scheme
+	controller       client.Object
+	resourceSpecHash string
+	pluginConfigHash string
+	data             v1alpha1.RbacAssessmentReportData
+}
+
+func NewReportBuilder(scheme *runtime.Scheme) *ReportBuilder {
+	return &ReportBuilder{
+		scheme: scheme,
+	}
+}
+
+func (b *ReportBuilder) Controller(controller client.Object) *ReportBuilder {
+	b.controller = controller
+	return b
+}
+
+func (b *ReportBuilder) ResourceSpecHash(hash string) *ReportBuilder {
+	b.resourceSpecHash = hash
+	return b
+}
+
+func (b *ReportBuilder) PluginConfigHash(hash string) *ReportBuilder {
+	b.pluginConfigHash = hash
+	return b
+}
+
+func (b *ReportBuilder) Data(data v1alpha1.RbacAssessmentReportData) *ReportBuilder {
+	b.data = data
+	return b
+}
+
+func (b *ReportBuilder) reportName() string {
```

Comment:
I know this works like other reports, but looks so weird for system roles, because we get a bunch of `cluster-<HASH>` roles, and to find the report for a specific role it is really hard, we need to go over one by one, eg:
```
clusterrole-54ccb57cc4                              Trivy     5h59m
clusterrole-54cdc9b678                              Trivy     5h59m
clusterrole-5585c7b9ff                              Trivy     5h59m
clusterrole-569d87574c                              Trivy     3h25m
clusterrole-575b7f6784                              Trivy     5h59m
clusterrole-57d745d4cc                              Trivy     5h59m
clusterrole-5857f84f59                              Trivy     5h59m
clusterrole-58bfc7788d                              Trivy     5h59m
clusterrole-5b458986c5                              Trivy     5h59m
clusterrole-5cbfdf6f9d                              Trivy     5h59m
clusterrole-5f9f8f6b4c                              Trivy     5h59m
....
```

They all belong to system roles, but really hard to find which one specifically.

```
system:aggregate-to-admin                                              2022-07-05T14:12:00Z
system:aggregate-to-edit                                               2022-07-05T14:12:00Z
system:aggregate-to-view                                               2022-07-05T14:12:00Z
system:auth-delegator                                                  2022-07-05T14:12:00Z
system:basic-user                                                      2022-07-05T14:12:00Z
system:certificates.k8s.io:certificatesigningrequests:nodeclient       2022-07-05T14:12:00Z
system:certificates.k8s.io:certificatesigningrequests:selfnodeclient   2022-07-05T14:12:00Z
system:certificates.k8s.io:kube-apiserver-client-approver              2022-07-05T14:12:00Z
system:certificates.k8s.io:kube-apiserver-client-kubelet-approver      2022-07-05T14:12:00Z
system:certificates.k8s.io:kubelet-serving-approver                    2022-07-05T14:12:00Z
system:certificates.k8s.io:legacy-unknown-approver                     2022-07-05T14:12:00Z
system:controller:attachdetach-controller                              2022-07-05T14:12:00Z
system:controller:certificate-controller                               2022-07-05T14:12:00Z
system:controller:clusterrole-aggregation-controller                   2022-07-05T14:12:00Z
...
```